microsoft/PowerToys
# [PowerToys Run] Plugin manager

## Summary of the Pull Request

**What is this about:**
![image](https://user-images.githubusercontent.com/17161067/108985975-53543200-769a-11eb-8fe9-0fc40a1f884a.png)

Specs: https://github.com/microsoft/PowerToys/wiki/PowerToys-Run-Plugin-spec

**What is include in the PR:** 
- Added `Name` and `Description` to `IPlugin` in order to take localized strings.
- Enable one action keyword for several plugins
- Move `disable drive detection warning' under the Windows Search plugin
- Plugins section UI 

**How does someone test / validate:** 
If you were testing PRs in `feature/plugin-manager` branch then delete `PowerToys Run\settings.json` file.

Test consistency of settings with the previous version: 
- Start PT Run for v0.31.2 and try to search for something
- Modify few settings in `PowerToys Run` tab in settings
- Start local build
- Check if settings persisted, PT Run works and there is plugins section on PowerToys Run tab
- Start PowerToys v0.31.2 and verify that settings persisted and PT Run works

Feature testing: 
- Check if settings are propagated to PT Run
- Try disabling a plugin, modifying `Include in global result` and `Direct activation phrase` for all plugins
- Switch the theme and check that the plugins section still looks okay
- Test tab navigation with the narrator. Take into account existing issue: 'Navigation to next or previous item does not work if the focus is on controls from AdditionalInfoPanel'
- Assign the same `Direct activation phrase` for several plugins and test it in PT Run

## Quality Checklist

- [X] **Linked issue:** #5273

## Contributor License Agreement (CLA)
A CLA must be signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/PowerToys) and sign the CLA.
